### ISSUE TITLE
[Task]: Support Cinder's job action webhook

### ISSUE THREAD

**Issue by eviljeff** [mozilla/addons#15035]
### Description

Cinder supports moving jobs between queues, both directly and as an escalation option (both are per-queue settings).  In general AMO doesn't care which queue an entity is in - we take actions based on the entity type.  Except for the AMO reviewer handled queue (the "content infringement queue") - if a job is in in that queue we don't take action on that back of decisions (or we'd be duplicating what was just carried out in the reviewer tools before notifying cinder of the decision).

If we support the job action webhook we can monitor for queue moves, unlocking the functionality to use Cinder to move jobs into ~and out of~ the reviewer handled queue, meaning moderators can use the Cinder escalation UI to escalate/forward cases to AMO rather than AMO_ESCALATE_ACTION.  

### Acceptance Criteria

```[tasklist]
  ### Milestones/checkpoints
  - [ ] queue moves are monitored and handled by AMO (using the existing webhook endpoint)
  - [ ] a move _into_ the AMO reviewer handled queue on cinder makes the job be marked as being handled in the reviewer tools, (forwarded), and creates the NeedsHumanReview flag to place it the AMO review queue

```

out of scope:
 - a move _out_ of the AMO reviewer handled queue in cinder.  If implemented the job would be un-marked as being handled in the reviewer tools, and the NeedsHumanReview flag cleared, but it's an edge case as no-one is supposed to be touching AMO reviewer tools handled jobs via cinder.
 - forwarding jobs from the reviewer tools to the legal escalation queue (or T&S, or others). - the job action webhook is only for notifying moves; we'd have to implement different API integration to _make_ queue moves.

### Checks

- [X] If I have identified that the work is specific to a repository, I have removed "repository:addons-server" or "repository:addons-frontend"

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/AMOENG-1076)

**Comment by diox** [comment]
Would simplify things both on our side and Cinder's (and potentially enable fancy tricks later) but is not a requirement for now.

**Comment by ioanarusiczki** [comment]
Looking at [str](https://github.com/mozilla/addons-server/pull/22795#issue-2614151581) I tried an extension report but also appeal. 

I've sent the report/appeal to Add-on Content Infringement queue which automatically flagged the versions for HR in AMO and made them visible in the Manual Review queue.  -> so far that's expected

I tried resolving the report and appeal from rev tools -> the HR flag is not removed. -> seems to be a problem 

Example with a report: https://reviewers.addons-dev.allizom.org/en-US/reviewers/review/633395
With an appeal resolved with force-enable from rev tools: https://reviewers.addons-dev.allizom.org/en-US/reviewers/review-listed/633328

I see emails being sent correctly.
Theme escalations still work as before from Escalate -> Escalate to AMO

**Comment by eviljeff** [comment]
@ioanarusiczki I think (🤞) the NHR problem will be fixed with #15115 / mozilla/addons-server#22799

**Comment by ioanarusiczki** [comment]
Ok, using the old flow , actually, I escalated a report to AMO Escalation and from there I applied Escalate-> Escalate to AMO.  I could resolve it and HR was removed
https://reviewers.addons-dev.allizom.org/en-US/reviewers/review/633329

(in case this scenario is a real use case, idk)
